src/app/courses/courses.component.ts

```diff
@@ -11,11 +11,28 @@ export class CoursesComponent implements OnInit {
 
   public items: ICourse[];
 
-  constructor(private cs: CourseService) { 
+  private _cs: CourseService;
+
+  public searchStr = '';
+
+  constructor(cs: CourseService) {
+    this._cs = cs;
```

Comment:
We don't need to define service because we use dependency injection mechanism (we inject service using constructor, so it's already defined)
```
constructor(private cs: CourseService){}
```
the code above would be enough to access to the service instance. 
and we don't need `private _cs: CourseService;` and `this._cs = cs;` parts

then we could use it via `this`, e.g. 
```
ngOnInit() {
    this.items = this.cs.findAll();
}
````

please rework this point.

Reply:
Changed according to your recommendations, but in the 3rd pull request. Didn't get used to this possibility of using constructor parameters directly